datastore/transaction_test.go

```diff
@@ -88,3 +88,398 @@ func TestNewTransaction(t *testing.T) {
+		ProjectId: mockProjectID,
+	}
+	beginTxnRes := &pb.BeginTransactionResponse{
+		Transaction: mockTxnID,
+	}
+
+	testcases := []struct {
+		desc      string
+		rpcInputs []addRPCInput
+		ops       []string
+		settings  *transactionSettings
+	}{
+		{
+			desc: "[Get, Get, Put, Commit] No options. First Get does not pass new_transaction",
+			rpcInputs: []addRPCInput{
+				{
+					wantReq: beginTxnReq,
+					resp:    beginTxnRes,
+				},
+				{
+					wantReq: lookupReqWithTxn,
+					resp:    lookupResWithTxn,
+				},
+				{
+					wantReq: lookupReqWithTxn,
+					resp:    lookupResWithTxn,
+				},
+				{
+					wantReq: commitReq,
+					resp:    commitRes,
+				},
+			},
+			ops:      []string{"Get", "Get", "Put", "Commit"},
+			settings: &transactionSettings{},
+		},
+		{
+			desc: "[Get, Get, Put, Commit] BeginLater. First Get passes new_transaction",
+			rpcInputs: []addRPCInput{
+				{
+					wantReq: lookupReqWithNewTxn,
+					resp:    lookupResWithNewTxn,
+				},
+				{
+					wantReq: lookupReqWithTxn,
+					resp:    lookupResWithTxn,
+				},
+				{
+					wantReq: commitReq,
+					resp:    commitRes,
+				},
+			},
+			ops:      []string{"Get", "Get", "Put", "Commit"},
+			settings: &transactionSettings{beginLater: true},
+		},
+		{
+			desc: "[RunQuery, Get, Put, Commit] No options. RunQuery does not pass new_transaction",
+			rpcInputs: []addRPCInput{
+				{
+					wantReq: beginTxnReq,
+					resp:    beginTxnRes,
+				},
+				{
+					wantReq: runQueryReqWithTxn,
+					resp:    runQueryResWithTxn,
+				},
+				{
+					wantReq: lookupReqWithTxn,
+					resp:    lookupResWithTxn,
+				},
+				{
+					wantReq: commitReq,
+					resp:    commitRes,
+				},
+			},
+			ops:      []string{"RunQuery", "Get", "Put", "Commit"},
+			settings: &transactionSettings{},
+		},
+		{
+			desc: "[RunQuery, Get, Put, Commit] BeginLater. RunQuery passes new_transaction",
+			rpcInputs: []addRPCInput{
+				{
+					wantReq: runQueryReqWithNewTxn,
+					resp:    runQueryResWithNewTxn,
+				},
+				{
+					wantReq: lookupReqWithTxn,
+					resp:    lookupResWithTxn,
+				},
+				{
+					wantReq: commitReq,
+					resp:    commitRes,
+				},
+			},
+			ops:      []string{"RunQuery", "Get", "Put", "Commit"},
+			settings: &transactionSettings{beginLater: true},
+		},
+		{
+			desc: "[RunAggregationQuery, Get, Put, Commit] No options. RunAggregationQuery does not pass new_transaction",
+			rpcInputs: []addRPCInput{
+				{
+					wantReq: beginTxnReq,
+					resp:    beginTxnRes,
+				},
+				{
+					wantReq: runAggQueryReqWithTxn,
+					resp:    runAggQueryResWithTxn,
+				},
+				{
+					wantReq: lookupReqWithTxn,
+					resp:    lookupResWithTxn,
+				},
+				{
+					wantReq: commitReq,
+					resp:    commitRes,
+				},
+			},
+			ops:      []string{"RunAggregationQuery", "Get", "Put", "Commit"},
+			settings: &transactionSettings{},
+		},
+		{
+			desc: "[RunAggregationQuery, Get, Put, Commit] BeginLater. RunAggregationQuery passes new_transaction",
```

Comment:
It looks like this test verifies that runAggregationQuery will pass in new transaction options. Just for my own knowledge though, what change in this PR allows newTransaction options to be passed in for the runAggregationQuery call? Does runAggregationQuery make a call to the new `parseReadOptions()`?

Reply:
RunAggregationQuery makes call to parseQueryReadOptions which calls parseReadOptions

 https://github.com/googleapis/google-cloud-go/pull/8972/files#diff-c04af33c4c45ff4606abe36bd859432effb278cca4146d3f818e540f0479fd8dR806